src/gpu/intel/jit/v2/conv/plan.cpp

```diff
@@ -261,60 +290,56 @@ class multiply_info_t {
 
     bool init_dpas(const layout_desc_t &a_desc, const layout_desc_t &b_desc,
             const layout_desc_t &c_desc) {
-        pvar_t m_dim;
-        pvar_t n_dim;
-        pvar_t k_dim;
+        fused_dim_t m_dim('m');
+        fused_dim_t n_dim('n');
+        fused_dim_t k_dim('k');
         for (auto &d : iter_tile_) {
             switch (to_bmnk(d)) {
-                case 'm':
-                    gpu_assert(m_dim.is_undef());
-                    m_dim = d;
-                    break;
-                case 'n':
-                    gpu_assert(n_dim.is_undef());
-                    n_dim = d;
-                    break;
-                case 'k':
-                    gpu_assert(k_dim.is_undef());
-                    k_dim = d;
-                    break;
+                case 'm': m_dim.add(d, iter_tile_[d]); break;
+                case 'n': n_dim.add(d, iter_tile_[d]); break;
+                case 'k': k_dim.add(d, iter_tile_[d]); break;
                 default: gpu_error_not_expected();
             }
         }
-        gpu_check(!m_dim.is_undef() && !n_dim.is_undef() && !k_dim.is_undef())
+        gpu_check(m_dim.ndims() == 1 && n_dim.ndims() == 1
+                && utils::one_of(k_dim.ndims(), 1, 2))
                 << "init_dpas: cannot initialize MNK dimensions.";
-        dim_t m_size = iter_tile_.at(m_dim);
-        dim_t n_size = iter_tile_.at(n_dim);
-        dim_t k_size = iter_tile_.at(k_dim);
+        if (k_dim.ndims() == 2) {
+            std::swap(k_dim.dims[0], k_dim.dims[1]);
+            std::swap(k_dim.sizes[0], k_dim.sizes[1]);
+        }
         uint8_t sdepth = 8;
         uint8_t rcount = 8;
         int type_size = a_type_.size();
-        gpu_check(m_size % rcount == 0)
-                << "init_dpas: M dimension size is invalid: " << m_size;
-        gpu_check(n_size % simd_ == 0)
-                << "init_dpas: N dimension size is invalid: " << n_size;
-        gpu_check((k_size * type_size) % (sdepth * 4) == 0)
-                << "init_dpas: K dimension size is invalid: " << k_size;
+        gpu_check(m_dim.size() % rcount == 0)
+                << "init_dpas: M dimension size is invalid: " << m_dim.size();
+        gpu_check(n_dim.size() % simd_ == 0)
+                << "init_dpas: N dimension size is invalid: " << n_dim.size();
+        gpu_check((k_dim.size() * type_size) % (sdepth * 4) == 0)
```

Comment:
[nitpick]
`const int ops_per_channel = 4` maybe?

Reply:
Thanks, added `dword_size`, "ops-per-channel" is not quite it as it's type-specific.